src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/intervener/IntervenersAboutToSubmitHandler.java

```diff
@@ -51,19 +57,28 @@ public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(Finrem
             selectedOperationCode, caseData.getIntervenersList().getValueCode(), caseId);
 
         switch (selectedOperationCode) {
-            case ADD_INTERVENER_ONE_CODE -> service.updateIntervenerOneDetails(callbackRequest);
-            case ADD_INTERVENER_TWO_CODE -> service.updateIntervenerTwoDetails(callbackRequest);
-            case ADD_INTERVENER_THREE_CODE -> service.updateIntervenerThreeDetails(callbackRequest);
-            case ADD_INTERVENER_FOUR_CODE -> service.updateIntervenerFourDetails(callbackRequest);
-            case DEL_INTERVENER_ONE_CODE -> service.removeIntervenerOneDetails(caseData, caseId);
-            case DEL_INTERVENER_TWO_CODE -> service.removeIntervenerTwoDetails(caseData, caseId);
-            case DEL_INTERVENER_THREE_CODE -> service.removeIntervenerThreeDetails(caseData, caseId);
-            case DEL_INTERVENER_FOUR_CODE -> service.removeIntervenerFourDetails(caseData, caseId);
+            case ADD_INTERVENER_ONE_CODE -> caseData.setCurrentIntervenerChangeDetails(service.updateIntervenerOneDetails(callbackRequest));
```

Comment:
what exactly this method doing ? what I understood - seems like determining if its add or remove or update - that can be determine in submit handler  based on selection  - seems this is not needed.

Reply:
based on discussions with yourself and ruban during initial implementation. I had originally made a submit handler but was merged into this handler so we can interact and save the generic of the intervener details so we have that context available to call the corresponder.